sonarts-core/src/rules/noAccessorFieldMismatchRule.ts

```diff
@@ -0,0 +1,200 @@
+        setterOrGetter.name,
+      );
+      accessorIsPublic = Walker.isPublic(accessor);
+    }
+    if (
+      accessorIsPublic &&
+      accessor.body &&
+      matchingField &&
+      Walker.bodyIsSimple(accessor.body, setterOrGetter.type) &&
+      !this.fieldIsUsed(accessor, matchingField)
+    ) {
+      this.addFailureAtNode(
+        accessor.name,
+        `Refactor this ${setterOrGetter.type} so that it actually refers to the property '${matchingField.name!.getText()}'`,
+      );
+    }
+  }
+
+  private static isPublic(method: ts.MethodDeclaration | ts.AccessorDeclaration): boolean {
+    const modifier = nav.accessModifier(method);
+    return !!modifier && modifier.kind === ts.SyntaxKind.PublicKeyword;
+  }
+
+  private static getName(accessor: ts.AccessorDeclaration) {
+    return accessor.name.getText().toLowerCase();
+  }
+
+  private setterOrGetter(method: ts.MethodDeclaration): { type: string; name: string } | undefined {
+    const name = method.name.getText().toLowerCase();
+    if (name.startsWith("set") || name.startsWith("Set")) {
+      if (method.parameters.length === 1) {
+        return { type: "setter", name: name.substring(3) };
+      }
+    }
+    if (name.startsWith("get") || name.startsWith("Get")) {
+      if (method.parameters.length === 0) {
+        return { type: "getter", name: name.substring(3) };
+      }
+    }
+  }
+
+  private static matchingField(
+    members: Array<ts.ClassElement | ts.ObjectLiteralElement>,
+    constructorDeclaredParameters: Array<ts.ParameterDeclaration>,
+    targetName: string,
+  ): Field | undefined {
+    return members
+      .filter(
+        element =>
+          element.kind === ts.SyntaxKind.PropertyDeclaration ||
+          element.kind === ts.SyntaxKind.PropertyAssignment ||
+          element.kind === ts.SyntaxKind.ShorthandPropertyAssignment ||
+          element.kind === ts.SyntaxKind.JsxAttribute,
+      )
+      .map(element => element as Field)
+      .concat(constructorDeclaredParameters)
+      .find(element => !!element.name && Walker.fieldNameMatches(element.name.getText(), targetName));
+  }
+
+  private static fieldNameMatches(fieldName: string, targetName: string): boolean {
+    const fieldNameLowerCase = fieldName.toLowerCase();
+    const underscoredTargetName = "_" + targetName;
+    return fieldNameLowerCase === targetName || fieldNameLowerCase === underscoredTargetName;
+  }
+
+  private static fieldsDeclaredInConstructorParameters(containingClass: ts.ClassDeclaration | ts.ClassExpression) {
+    const constr = nav.constructorOf(containingClass);
+    if (constr) {
+      return constr.parameters.filter(parameter => nav.accessModifier(parameter) || nav.isReadonly(parameter));
+    } else {
+      return [];
+    }
+  }
+
+  private static bodyIsSimple(body: ts.Block, type: string): boolean {
+    if (body.statements.length > 1) {
+      return false;
+    }
+    if (body.statements.length == 0) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') == should be === [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1440)